src/opnsense/mvc/app/library/OPNsense/Routing/Gateways.php

```diff
@@ -234,41 +235,38 @@ public function getGateways()
                 'is_loopback' => true
             ];
             // iterate configured gateways
-            if (!empty($this->configHandle->gateways)) {
-                foreach ($this->configHandle->gateways->children() as $tag => $gateway) {
-                    if ($tag == "gateway_item" && !empty($gateway)) {
-                        if (in_array((string)$gateway->name, $reservednames)) {
-                            syslog(LOG_WARNING, 'Gateway: duplicated entry "' . $gateway->name . '" in config.xml needs manual removal');
-                        }
-                        $reservednames[] = (string)$gateway->name;
-                        $gw_arr = array();
-                        foreach ($gateway as $key => $value) {
-                            $gw_arr[(string)$key] = (string)$value;
-                        }
-                        if (empty($gw_arr['priority'])) {
-                            // default priority
-                            $gw_arr['priority'] = 255;
-                        }
-                        if (empty($gw_arr['ipprotocol'])) {
-                            // default address family
-                            $gw_arr['ipprotocol'] = 'inet';
+            foreach ((new Gateway())->gatewayIterator() as $gw_arr) {
+                if (!empty($gw_arr)) {
```

Comment:
Is it possible for `gatewayIterator()` to return empty arrays ? That looks a bit problematic

Reply:
No, not possible :)